src/components/autocomplete.js

```diff
@@ -26,18 +26,18 @@ const Autocomplete = () => {
   const [activeSuggestionIndex, setActiveSuggestionIndex] = useState(0);
   const [getCatBreeds, { loading, data }] = useLazyQuery(GET_BREED);
 
-  function changeHandler(query) {
-    if (!query) return setSuggestions([]);
-    setInput(query);
-
-    const debouncedChangeFilter = debounce(() => {
+  const debouncedApiCall = useCallback(
+    debounce((text) => {
       getCatBreeds({
-        variables: { breedName: query }
+        variables: { breedName: text }
       });
-    }, 500);
+    }, 500),
+    []
+  );
 
-    debouncedChangeFilter();
-  }
+  useEffect(() => {
+    debouncedApiCall(input);
+  }, [input, debouncedApiCall]);
```

Comment:
It's a good idea to clean-up in case this component gets unmounted.

You can use `.cancel`: https://lodash.com/docs/4.17.15#debounce